src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/resources/viewmodel/ResourcesViewModel.kt

```diff
@@ -1,43 +1,48 @@
 package org.wycliffeassociates.otter.jvm.app.ui.resources.viewmodel
 
 import javafx.application.Platform
-import javafx.beans.property.SimpleObjectProperty
-import javafx.beans.property.SimpleStringProperty
 import org.wycliffeassociates.otter.common.data.workbook.*
+import org.wycliffeassociates.otter.common.domain.content.Recordable
 import org.wycliffeassociates.otter.common.utils.mapNotNull
+import org.wycliffeassociates.otter.jvm.app.ui.resourcetakes.viewmodel.TakesViewModel
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
 import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.ResourceGroupCardItemList
 import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.resourceGroupCardItem
 import tornadofx.*
 
 class ResourcesViewModel : ViewModel() {
-    val activeWorkbookProperty = SimpleObjectProperty<Workbook>()
-    val workbook: Workbook
-        get() = activeWorkbookProperty.value
+    private val takesViewModel: TakesViewModel by inject()
+    private val workbookViewModel: WorkbookViewModel by inject()
 
-    val activeChapterProperty = SimpleObjectProperty<Chapter>()
-    val chapter: Chapter
-        get() = activeChapterProperty.value
-
-    val activeResourceSlugProperty = SimpleStringProperty()
-    val resourceSlug: String
-        get() = activeResourceSlugProperty.value
-
-    val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList(mutableListOf())
+    val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList()
 
     fun loadResourceGroups() {
-        chapter
-            .children
-            .startWith(chapter)
-            .mapNotNull { resourceGroupCardItem(it, resourceSlug, onSelect = this::navigateToTakesPage) }
-            .buffer(2) // Buffering by 2 prevents the list UI from jumping while groups are loading
-            .subscribe {
-                Platform.runLater {
-                    resourceGroups.addAll(it)
+        workbookViewModel.chapter.let { chapter ->
+            chapter
+                .children
+                .startWith(chapter)
+                .mapNotNull {
+                    resourceGroupCardItem(it, workbookViewModel.resourceSlug, onSelect = this::navigateToTakesPage)
+                }
+                .buffer(2) // Buffering by 2 prevents the list UI from jumping while groups are loading
+                .subscribe {
+                    Platform.runLater {
+                        resourceGroups.addAll(it)
+                    }
```

Comment:
This should probably happen in the viewcontroller

Reply:
Isn't the observable list usually contained and manipulated in the view model, and the view/fragment simply listens for changes to the list?

Reply:
So it turns out that we don't need to use Platform.runLater when we are adding items to an observable list. The listview will automatically notice changes to the observable list and add the items anyway. I was confused because in the case where resource cards are added to the resource group card, we are actually directly adding a node to the node graph- in which case we need Platform.runLater.

By removing Platform.runLater, we're closer to being able to move this ViewModel into common!